### DISCUSSION TITLE
Uncertainty Quantification Scripting

### DISCUSSION THREAD

**Discussion by RachelETHOMAS** [simnibs/simnibs#168] [2. General]
Hi there,

I'm trying to incorporate UQ script into my working matlab script for COND values in SimNIBS. I have tried using my script on both my own data set and the 'ernie' data set, and in both cases the same error message appears after performing simulation #16? The message reads (from#16):

[ simnibs ]INFO: Performing simulation #16
[ simnibs ]INFO: Using isotropic conductivities
[ simnibs ]INFO: Simulating electrode pair 2101 - 2102
[ simnibs ]INFO: Assembling FEM Matrx
[ simnibs ]INFO: 50.34s to assemble FEM matrix
[ simnibs ]INFO: Using solver options: -ksp_type cg -ksp_rtol 1e-10 -pc_type hypre -pc_hypre_type boomeramg -pc_hypre_boomeramg_coarsen_type HMIS
[ simnibs ]INFO: Preparing the KSP
[ simnibs ]INFO: Time to prepare the KSP: 4.01s
[ simnibs ]INFO: Solving system 1 of 1
[ simnibs ]INFO: Running PETSc with KSP: cg PC: hypre
[ simnibs ]INFO: Number of iterations: 28 Residual Norm: 5.32e-09
[ simnibs ]INFO: KSP Converged with reason: 2
[ simnibs ]INFO: Time to solve system: 14.55s
[ simnibs ]INFO: Estimated current calibration error: 0.3%
[ simnibs ]INFO: gPC Error Estimate: 9.065259e-03
[ simnibs ]INFO: Calculating fields: e
[ simnibs ]INFO: Expanding field: magnE
[ simnibs ]INFO: Estimated error: 8.653166e-03 for field magnE
[ simnibs ]INFO: Expanding field: E
[ simnibs ]CRITICAL: Uncaught exception
ValueError: Unable to create dataset (name already exists)

Error in simnibs_script_ernie_UQ (line 150)
run_simnibs(s)

I've attached below the script I'm using, please do you have any advice/solutions for how I may fix this issue?

Thanks very much,
Rachel

file:///cubric/data/c2021194/SIMNIBS_CONDUC_TEST/anat/S03/S03_motor_montage/simnibs_script_S03_motorMontage_UQ.m.zip

**Comment by oulap** [comment]
Hi,

I don't think the attachment got added. Could you send the script that fails on ernie, so I can try to reproduce?

Thanks

**Comment by RachelETHOMAS** [comment]
[simnibs_script_S03_motorMontage_UQ.m.zip](https://github.com/simnibs/simnibs/files/12051361/simnibs_script_S03_motorMontage_UQ.m.zip)

Hi,
Sorry hopefully this has worked. 

Thanks in advance!

**Comment by RachelETHOMAS** [comment]
*The paths to the file aren't currently set up for ernie so will need to be edited...

**Comment by RachelETHOMAS** [comment]
Hi there,

I was just wondering if any progress had been made with this - I'm still trying to incorporate it into my work and any advice regarding how to address this issue, or any information regarding the differences in environments would be appreciated :)

**Reply by oulap** [reply]
Sorry for the lack of updates. This is indeed a bug, even the standard example fails. We're currently trying to figure out how to best solve it.

**Reply by RachelETHOMAS** [reply]
Awesome thanks!
